### PR TITLE
[Project] Refactor README.md

### PR DESCRIPTION
**Pull request by Tim2313**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,59 @@
...
+
+- **SIZE** - print size at specified price (bid, ask or spread).
+
+- **BUY** - removes <size> shares out of asks, most cheap ones.
+
+- **SELL** - removes <size> shares out of bids, most expensive ones.
+
+### Examples:
+
+- **u,<price>,<size>,bid** - set bid size at <price> to <size>.
```
Comment (MLGMAG) [inline README.md:27]: Add `\` before '<' and '>'.

File: README.md [hunk 2]
```diff
@@ -1,3 +1,59 @@
-## My test task
+# My test task
+
+---
 
 Here you can find my implementation of OrderBook.
+
+---
```
Comment (MLGMAG) [inline README.md:7]: Add a terminology section.
Like:
```
## Terminology
- Ask price -> ...
- Bid price -> ...
```

File: README.md [hunk 3]
```diff
@@ -1,3 +1,59 @@
...
+
+## Type of operations: Update, Order, Query. Their commands:##
```
Comment (MLGMAG) [inline README.md:9]: Extract types into separate section:
```
## Type of operations
- Update -> `u`
- Order -> `o`
- Query -> `q`
```

File: README.md [hunk 4]
```diff
@@ -1,3 +1,59 @@
-## My test task
+# My test task
```
Comment (MLGMAG) [inline README.md:1]: Change the title to `OrderBook`

File: README.md [hunk 5]
```diff
@@ -1,3 +1,59 @@
...
+
+- **q,size,<price>** - print size at specified price (bid, ask or spread).
+
+- **o,buy,<size>** - removes <size> shares out of asks, most cheap ones.
+
+- **o,sell,<size>** - removes <size> shares out of bids, most expensive ones
+
+## Examples of input file:
+
+```html
```
Comment (MLGMAG) [inline README.md:43]: Remove `html`

File: README.md (hunk 2 above)
Comment (MLGMAG) [inline README.md:7]: Add the `Background` section that will explain the real world problem.
And show, how you solution fix it.

**Review by MLGMAG — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (MLGMAG) [inline README.md:7]: Add build section.
This section will explain, how to build your project

Comment (MLGMAG) [reply README.md:7]: Add `Pre-requisites`.
This section will describe the technologies that should be installed on the OS to build/run your program.

Comment (Tim2313) [reply README.md:31]: fix

File: README.md (hunk 3 above)
Comment (Tim2313) [reply README.md]: fix

File: README.md (hunk 2 above)
Comment (Tim2313) [reply README.md:31]: fix

File: buildJar.bash [hunk 6]
```diff
@@ -18,3 +18,9 @@ jar cfm TestTaskBook.jar ../$MANIFEST_FILE ./*
 
 mv TestTaskBook.jar ../
 
+
```
Comment (MLGMAG) [inline buildJar.bash:21]: Remove empty lines

File: README.md [hunk 7]
```diff
@@ -1,3 +1,91 @@
-## My test task
+# OrderBook
+
+---
 
 Here you can find my implementation of OrderBook.
+
+## Background:
```
Comment (MLGMAG) [inline README.md:7]: Add new lines to the 'Background' section.
image

File: README.md [hunk 8]
```diff
@@ -1,3 +1,91 @@
...
+To build and compile the project, follow these steps:
+- You need linux OS;
+- Open a terminal or command prompt and navigate to the root directory of your Maven project;
+- Run the following command to build the JAR file **mvn clean package**;
+- Then run the following command **java -jar target/your-project.jar**
+
+## Pre-requisites:
+Before building and running the program, make sure 
+you have the following technologies installed on your operating system:
+- Java Development Kit (JDK);
```
Comment (MLGMAG) [inline README.md:27]: Replace JDK with 'Java <version>'

**Review by MLGMAG — CHANGES_REQUESTED** [review]

File: buildJar.bash (hunk 6 above)
Comment (Tim2313) [reply buildJar.bash:21]: fix

File: README.md [hunk 9]
```diff
@@ -1,3 +1,91 @@
...
+
+- **q,best_bid** - print best bid price and size.
+
+- **q,best_ask** - print best ask price and size.
+
+- **q,size,\<price>\** - print size at specified price (bid, ask or spread).
+
+- **o,buy,\<size>\** - removes <size> shares out of asks, most cheap ones.
+
+- **o,sell,\<size>\** - removes <size> shares out of bids, most expensive ones.
```
[the remark this answers is not in the document]
Comment (Tim2313) [reply README.md:71]: fix

File: README.md (hunk 8 above)
Comment (Tim2313) [reply README.md:27]: fix

File: README.md [hunk 10]
```diff
@@ -1,3 +1,91 @@
...
+- Open a terminal or command prompt and navigate to the root directory of your Maven project;
+- Run the following command to build the JAR file **mvn clean package**;
+- Then run the following command **java -jar target/your-project.jar**
+
+## Pre-requisites:
+Before building and running the program, make sure 
+you have the following technologies installed on your operating system:
+- Java Development Kit (JDK);
+- Maven;
+- Linux OS;
```
[the remark this answers is not in the document]
Comment (Tim2313) [reply README.md:29]: fix

File: README.md [hunk 11]
```diff
@@ -1,3 +1,91 @@
...
+## Type of operations:
+- Update -> `u`.
+- Order -> `o`.
+- Query -> `q`.
+
+## Commands of the operations: 
+
+- **BID** - bid.
+
+- **ASK** - ask.
```
[the remark this answers is not in the document]
Comment (Tim2313) [reply README.md:45]: fix

File: README.md [hunk 12]
```diff
@@ -1,3 +1,91 @@
...
+  u,11,5,ask
+  q,best_bid
+  u,10,2,bid
+  q,best_bid
+  o,sell,1
+  q,size,10
+  u,9,0,bid
+  u,11,0,ask
+
+## Examples of output file:
```
[the remark this answers is not in the document]
Comment (Tim2313) [reply README.md:86]: fix

File: README.md [hunk 13]
```diff
@@ -1,3 +1,96 @@
...
+
+- **BEST_ASK** - print best ask price and size.
+
+- **SIZE** - print size at specified price (bid, ask or spread).
+
+- **BUY** - removes size shares out of asks, most cheap ones.
+
+- **SELL** - removes size shares out of bids, most expensive ones.
+
+### Examples:
```
Comment (MLGMAG) [inline README.md:62]: You repeat word 'example' many times.

Replace this title with 'Commands usage'.

File: README.md [hunk 14]
```diff
@@ -1,3 +1,96 @@
...
+
+- **q,best_ask** - print best ask price and size.
+
+- **q,size,\<price>** - print size at specified price (bid, ask or spread).
+
+- **o,buy,\<size>** - removes <size> shares out of asks, most cheap ones.
+
+- **o,sell,\<size>** - removes <size> shares out of bids, most expensive ones.
+
+## Examples of input file:
```
Comment (MLGMAG) [inline README.md:78]: Place input/output files under one section: 'Example'

Under the general section, create sub sections 'input file' and 'output file'

File: buildJar.bash (hunk 6 above)
Comment (MLGMAG) [reply buildJar.bash:21]: The blank lines still present...

File: README.md [hunk 15]
```diff
@@ -1,3 +1,96 @@
...
+  u,10,2,bid
+  q,best_bid
+  o,sell,1
+  q,size,10
+  u,9,0,bid
+  u,11,0,ask
+```
+## Examples of output file:
+```
+
```
Comment (MLGMAG) [inline README.md:92]: Remove empty line

File: README.md [hunk 16]
```diff
@@ -1,3 +1,96 @@
...
+is always going to perform the trade and then leaves the market.
+In other words, each price level (for simplicity let's think of it as an integer value) can be either 
+bid (there are people willing to buy at this price), 
+ask (people are willing to sell at this price) or spread (nobody is willing to buy or sell at this price).
+Generally, the order book looks like the following example (B - bid, S - spread, A - ask), 
+size defines how many shares can be bought/sold at this price:
+
+## Build:
+To build and compile the project, follow these steps:
+- You need linux OS;
```
Comment (MLGMAG) [inline README.md:28]: You have the same line in the `Pre-requisites` section

File: README.md (hunk 7 above)
Comment (MLGMAG) [reply README.md:7]: Please, fix it

File: README.md [hunk 17]
```diff
@@ -1,3 +1,96 @@
...
+Those "limit orders" (people willing to buy/sell in certain quantities) are the limit order book. 
+In some cases people are willing to buy/sell at any price (that's called a "market order"), such a person 
+is always going to perform the trade and then leaves the market.
+In other words, each price level (for simplicity let's think of it as an integer value) can be either 
+bid (there are people willing to buy at this price), 
+ask (people are willing to sell at this price) or spread (nobody is willing to buy or sell at this price).
+Generally, the order book looks like the following example (B - bid, S - spread, A - ask), 
+size defines how many shares can be bought/sold at this price:
+
+## Build:
```
Comment (MLGMAG) [inline README.md:26]: Usually the column symbol `:` is not added to the titles.
Remove them

**Review by MLGMAG — CHANGES_REQUESTED** [review]

File: README.md (hunk 17 above)
Comment (Tim2313) [reply README.md]: fix

File: README.md (hunk 16 above)
Comment (Tim2313) [reply README.md:29]: fix

File: README.md (hunk 15 above)
Comment (Tim2313) [reply README.md]: fix

File: README.md (hunk 14 above)
Comment (Tim2313) [reply README.md]: fix

File: README.md (hunk 13 above)
Comment (Tim2313) [reply README.md]: fix

File: README.md [hunk 18]
```diff
@@ -1,3 +1,99 @@
...
+Generally, the order book looks like the following example (B - bid, S - spread, A - ask), 
+size defines how many shares can be bought/sold at this price:
+
+## Build
+
+To build and compile the project, follow these steps:
+- You need linux OS;
+- Open a terminal or command prompt and navigate to the root directory of your Maven project;
+- Run the following command to build the JAR file **mvn clean package**;
+- Then run the following command **java -jar target/your-project.jar**
```
Comment (MLGMAG) [inline README.md:32]: The CLI commands should be displayed as code

File: README.md [hunk 19]
```diff
@@ -1,3 +1,99 @@
...
+
+- **q,best_bid** - print best bid price and size.
+
+- **q,best_ask** - print best ask price and size.
+
+- **q,size,\<price>** - print size at specified price (bid, ask or spread).
+
+- **o,buy,\<size>** - removes <size> shares out of asks, most cheap ones.
+
+- **o,sell,\<size>** - removes <size> shares out of bids, most expensive ones.
```
Comment (MLGMAG) [inline README.md:78]: Some things is displayed wrongly
image

File: README.md [hunk 20]
```diff
@@ -1,3 +1,99 @@
...
+
+Before building and running the program, make sure 
+you have the following technologies installed on your operating system:
+- Java;
+- Maven;
+
+## Terminology
+
+- Ask price -> price of the ask.
+- Bid price -> price of the bid.
```
Comment (MLGMAG) [inline README.md:44]: Ask is ask... Bid is bid... Add more meaningful description.

File: README.md [hunk 21]
```diff
@@ -1,3 +1,99 @@
...
+- Open a terminal or command prompt and navigate to the root directory of your Maven project;
+- Run the following command to build the JAR file **mvn clean package**;
+- Then run the following command **java -jar target/your-project.jar**
+
+## Pre-requisites
+
+Before building and running the program, make sure 
+you have the following technologies installed on your operating system:
+- Java;
+- Maven;
```
Comment (MLGMAG) [inline README.md:39]: Remove `;`

**Review by MLGMAG — CHANGES_REQUESTED** [review]

File: README.md (hunk 7 above)
Comment (Tim2313) [reply README.md]: fix

Comment (MLGMAG) [reply README.md]: There is still no new lines. Please add them, because text is unreadable.

File: buildJar.bash (hunk 6 above)
Comment (Tim2313) [reply buildJar.bash]: fix

File: README.md (hunk 7 above)
Comment (Tim2313) [reply README.md:7]: fix

File: README.md (hunk 18 above)
Comment (Tim2313) [reply README.md]: fix

File: README.md (hunk 19 above)
Comment (Tim2313) [reply README.md]: fix

File: README.md (hunk 21 above)
Comment (Tim2313) [reply README.md]: fix

File: README.md (hunk 20 above)
Comment (Tim2313) [reply README.md]: fix

File: README.md [hunk 22]
```diff
@@ -1,3 +1,94 @@
...
+- Ask price -> The "Ask" refers to the price level at which sellers are willing to sell their shares. 
+It represents the minimum price at which someone is willing to part with their shares.
+
+- Bid price -> On the other hand, the "Bid" represents the price level at which buyers are willing to buy shares. 
+It represents the maximum price that a buyer is willing to pay for the shares.
+
+## Type of operations:
+- Update -> `u`.
+- Order -> `o`.
+- Query -> `q`.
```
Comment (MLGMAG) [inline README.md:49]: Write this as a table

File: README.md [hunk 23]
```diff
@@ -1,3 +1,94 @@
...
+- Maven
+- Operations systems based on linux
+
+---
+## Terminology:
+- Ask price -> The "Ask" refers to the price level at which sellers are willing to sell their shares. 
+It represents the minimum price at which someone is willing to part with their shares.
+
+- Bid price -> On the other hand, the "Bid" represents the price level at which buyers are willing to buy shares. 
+It represents the maximum price that a buyer is willing to pay for the shares.
```
Comment (MLGMAG) [inline README.md:44]: Replace `->` with `-`

File: README.md [hunk 24]
```diff
@@ -1,3 +1,94 @@
...
+In other words, each price level (for simplicity, let's think of it as an integer value) can be either bid (there are people willing to buy at this price), ask (people are willing to sell at this price), or spread (nobody is willing to buy or sell at this price).
+
+Generally, the order book looks like the following example (B - bid, S - spread, A - ask). The size defines how many shares can be bought/sold at this price:
+
+## Build:
+To build and compile the project, follow these steps:
+- You need linux OS
+- Open a terminal or command prompt and navigate to the root directory of your Maven project ```cd TestTaskProxyBand```
+- Run the following command to build the JAR file ```mvn clean package```
+- Then run the following command ```java -jar target/your-project.jar```
```
Comment (MLGMAG) [inline README.md:29]: Add `:` before command

File: README.md [hunk 25]
```diff
@@ -1,3 +1,94 @@
...
+
+In some cases, people are willing to buy/sell at any price (that's called a "market order"). Such a person is always going to perform the trade and then leaves the market.
+
+In other words, each price level (for simplicity, let's think of it as an integer value) can be either bid (there are people willing to buy at this price), ask (people are willing to sell at this price), or spread (nobody is willing to buy or sell at this price).
+
+Generally, the order book looks like the following example (B - bid, S - spread, A - ask). The size defines how many shares can be bought/sold at this price:
+
+## Build:
+To build and compile the project, follow these steps:
+- You need linux OS
```
Comment (MLGMAG) [inline README.md:26]: Remove this line. We are mentioning this in the `Pre-requisites` section

File: README.md [hunk 26]
```diff
@@ -1,3 +1,94 @@
-## My test task
+# OrderBook
+
+---
 
 Here you can find my implementation of OrderBook.
```
Comment (MLGMAG) [inline README.md:5]: Remove this line

**Review by MLGMAG — CHANGES_REQUESTED** [review]

File: README.md (hunk 23 above)
Comment (Tim2313) [reply README.md:44]: fix

File: README.md (hunk 24 above)
Comment (Tim2313) [reply README.md:29]: fix

File: README.md (hunk 26 above)
Comment (Tim2313) [reply README.md:5]: fix

File: README.md (hunk 25 above)
Comment (Tim2313) [reply README.md:26]: fix

File: README.md (hunk 22 above)
Comment (Tim2313) [reply README.md:49]: fix

File: README.md (hunk 26 above)
Comment (MLGMAG) [reply README.md:5]: Line `Here you can find my implementation of OrderBook.` is still present...

File: README.md [hunk 27]
```diff
@@ -1,3 +1,95 @@
...
+
+- Bid price - On the other hand, the "Bid" represents the price level at which buyers are willing to buy shares. 
+It represents the maximum price that a buyer is willing to pay for the shares.
+
+## Type of operations:
+| Name   | key |
+|--------|-----|
+| Update | 'u' |
+| Order  | 'o' |
+| Query  | 'q' |
```
Comment (MLGMAG) [inline README.md:50]: Remove `'` symbol.
Capitalize word `key`

File: README.md [hunk 28]
```diff
@@ -1,3 +1,95 @@
...
+
+In some cases, people are willing to buy/sell at any price (that's called a "market order"). Such a person is always going to perform the trade and then leaves the market.
+
+In other words, each price level (for simplicity, let's think of it as an integer value) can be either bid (there are people willing to buy at this price), ask (people are willing to sell at this price), or spread (nobody is willing to buy or sell at this price).
+
+Generally, the order book looks like the following example (B - bid, S - spread, A - ask). The size defines how many shares can be bought/sold at this price:
+
+## Build:
+To build and compile the project, follow these steps:
+- Open a terminal or command prompt and navigate to the root directory of your Maven project: ```cd TestTaskProxyBand```
```
Comment (MLGMAG) [inline README.md:26]: `your Maven` why `your`?

Rewrite it like:
```
Open a terminal and navigate to the root directory of the project: ...
```

File: README.md [hunk 29]
```diff
@@ -1,3 +1,95 @@
...
+- Run the following command to build the JAR file: ```mvn clean package```
+- Then run the following command: ```java -jar target/your-project.jar```
+
+## Pre-requisites:
+Before building and running the program, make sure 
+you have the following technologies installed on your operating system:
```
Comment (MLGMAG) [inline README.md:32]: Extend it like this:
```
... on your Linux based operating system
``` 

And remove `Operations systems based on linux` item

File: README.md [hunk 30]
```diff
@@ -1,3 +1,95 @@
...
+- Java
+- Maven
+- Operations systems based on linux
```
Comment (MLGMAG) [inline README.md:35]: Specify the version of the mentioned technologies

File: README.md [hunk 31]
```diff
@@ -1,3 +1,95 @@
...
+In some cases, people are willing to buy/sell at any price (that's called a "market order"). Such a person is always going to perform the trade and then leaves the market.
+
+In other words, each price level (for simplicity, let's think of it as an integer value) can be either bid (there are people willing to buy at this price), ask (people are willing to sell at this price), or spread (nobody is willing to buy or sell at this price).
+
+Generally, the order book looks like the following example (B - bid, S - spread, A - ask). The size defines how many shares can be bought/sold at this price:
+
+## Build:
+To build and compile the project, follow these steps:
+- Open a terminal or command prompt and navigate to the root directory of your Maven project: ```cd TestTaskProxyBand```
+- Run the following command to build the JAR file: ```mvn clean package```
```
Comment (MLGMAG) [inline README.md:27]: I thing one '`' symbol is enough

**Review by MLGMAG — CHANGES_REQUESTED** [review]

File: README.md (hunk 27 above)
Comment (Tim2313) [reply README.md:50]: fix

File: README.md (hunk 28 above)
Comment (Tim2313) [reply README.md:26]: fix

File: README.md [hunk 32]
```diff
@@ -1,3 +1,95 @@
...
+- Then run the following command: ```java -jar target/your-project.jar```
```
[the remark this answers is not in the document]
Comment (Tim2313) [reply README.md:28]: fix

File: README.md (hunk 29 above)
Comment (Tim2313) [reply README.md:32]: fix

File: README.md (hunk 30 above)
Comment (Tim2313) [reply README.md:35]: fix

File: README.md (hunk 31 above)
Comment (Tim2313) [reply README.md:27]: fix

File: README.md [hunk 33]
```diff
@@ -1,3 +1,90 @@
...
+To build and compile the project, follow these steps:
+- Open a terminal and navigate to the root directory of the project:`cd TestTaskProxyBand`
+- Run the following command to build the JAR file: `mvn clean package`
+- Then start the program with the following command.: `java -jar <your-project>.jar`
+
+## Pre-requisites:
+Before building and running the program, make sure 
+you have the following technologies installed on your Linux based operating system:
+- Java (recommended version **11**)
+- Maven (recommended version **3.2**)
```
Comment (MLGMAG) [inline README.md:30]: Add only version like:
```
Java 11
```

File: README.md [hunk 34]
```diff
@@ -1,3 +1,90 @@
...
+
+In other words, each price level (for simplicity, let's think of it as an integer value) can be either bid (there are people willing to buy at this price), ask (people are willing to sell at this price), or spread (nobody is willing to buy or sell at this price).
+
+Generally, the order book looks like the following example (B - bid, S - spread, A - ask). The size defines how many shares can be bought/sold at this price:
+
+## Build:
+To build and compile the project, follow these steps:
+- Open a terminal and navigate to the root directory of the project:`cd TestTaskProxyBand`
+- Run the following command to build the JAR file: `mvn clean package`
+- Then start the program with the following command.: `java -jar <your-project>.jar`
```
Comment (MLGMAG) [inline README.md:24]: Remove extra dot `.`

**Review by MLGMAG — CHANGES_REQUESTED** [review]

File: README.md (hunk 32 above)
[the remark this answers is not in the document]
Comment (MLGMAG) [reply README.md]: I thought, we should add an argument? (Am I right?)

File: README.md (hunk 33 above)
Comment (Tim2313) [reply README.md]: fix

File: README.md (hunk 34 above)
Comment (Tim2313) [reply README.md]: fix

**Review by MLGMAG — CHANGES_REQUESTED** [review]

File: README.md [hunk 35]
```diff
@@ -1,3 +1,90 @@
...
+
+In other words, each price level (for simplicity, let's think of it as an integer value) can be either bid (there are people willing to buy at this price), ask (people are willing to sell at this price), or spread (nobody is willing to buy or sell at this price).
+
+Generally, the order book looks like the following example (B - bid, S - spread, A - ask). The size defines how many shares can be bought/sold at this price:
+
+## Build:
+To build and compile the project, follow these steps:
+- Open a terminal and navigate to the root directory of the project:`cd TestTaskProxyBand`
+- Run the following command to build the JAR file: `mvn clean package`
+- Then start the program with the following command: `java -jar <your-project>.jar`
```
[the remark this answers is not in the document]
Comment (Tim2313) [reply README.md]: fix

File: README.md [hunk 36]
```diff
@@ -1,3 +1,90 @@
...
+
```
[the remark this answers is not in the document]
Comment (Tim2313) [reply README.md:27]: fix

File: README.md [hunk 37]
```diff
@@ -21,7 +21,9 @@ Generally, the order book looks like the following example (B - bid, S - spread,
 To build and compile the project, follow these steps:
 - Open a terminal and navigate to the root directory of the project:`cd TestTaskProxyBand`
 - Run the following command to build the JAR file: `mvn clean package`
-- Then start the program with the following command: `java -jar <your-project>.jar`
+
+## Usage
+- To start the program with the following command: `java -jar <your-project>.jar`
```
Comment (MLGMAG) [inline README.md:26]: Replace `start` with `execute`.

Comment (MLGMAG) [inline README.md:26]: Remove the list here, because it consists of only 1 element

Comment (MLGMAG) [inline README.md:26]: Add information about input file. (name, location)
Explain the program output for the following flows:

- If there is no input file
- If input file is present

**Review by MLGMAG — CHANGES_REQUESTED** [review]

File: README.md (hunk 37 above)
Comment (Tim2313) [reply README.md]: fix

Comment (Tim2313) [reply README.md]: fix

Comment (Tim2313) [reply README.md]: fix

File: README.md [hunk 38]
```diff
@@ -23,7 +23,14 @@ To build and compile the project, follow these steps:
 - Run the following command to build the JAR file: `mvn clean package`
 
 ## Usage
-- To start the program with the following command: `java -jar <your-project>.jar`
+You need to copy input file to the same directory which you move your project.jar.
+To execute the program with the following command: `java -jar <your-project>.jar`.
```
Comment (MLGMAG) [inline README.md:27]: Change to:
```
Execute the program with help of the following command:
```

Comment (MLGMAG) [inline README.md:27]: Change `<your-project>.jar` to `<jar_title>.jar`

File: README.md [hunk 39]
```diff
@@ -23,7 +23,14 @@ To build and compile the project, follow these steps:
 - Run the following command to build the JAR file: `mvn clean package`
 
 ## Usage
-- To start the program with the following command: `java -jar <your-project>.jar`
+You need to copy input file to the same directory which you move your project.jar.
```
Comment (MLGMAG) [inline README.md:26]: It important to use passive voice for writing the technical docs.
Do not use `You`.

Rewrite it like:
`The input file should be located in the same directory with the JAR file.`

**Review by MLGMAG — CHANGES_REQUESTED** [review]

File: README.md (hunk 37 above)
Comment (MLGMAG) [reply README.md]: Add information about the input file `title`.

File: README.md (hunk 38 above)
Comment (Tim2313) [reply README.md:27]: fix

Comment (Tim2313) [reply README.md:27]: fix

File: README.md (hunk 39 above)
Comment (Tim2313) [reply README.md:26]: fix

File: README.md (hunk 37 above)
Comment (Tim2313) [reply README.md]: fix

File: README.md [hunk 40]
```diff
@@ -23,7 +23,15 @@ To build and compile the project, follow these steps:
 - Run the following command to build the JAR file: `mvn clean package`
 
 ## Usage
-- To start the program with the following command: `java -jar <your-project>.jar`
+The input file should contain only commands from block **Examples of syntax**.
```
Comment (MLGMAG) [inline README.md:26]: Replace `block` with `section`.
And wrap the section title with double quotes `"`.

File: README.md [hunk 41]
```diff
@@ -23,7 +23,15 @@ To build and compile the project, follow these steps:
...
+The input file should be located in the same directory with the JAR file.
+Execute the program with help of the following command: `java -jar <jar_title>.jar`.
+
```
Comment (MLGMAG) [inline README.md:29]: Add line:
```
You will get the following results based on the input:
```

**Review by MLGMAG — CHANGES_REQUESTED** [review]

File: README.md (hunk 37 above)
Comment (MLGMAG) [reply README.md]: I still don't see the information about the input file `title` ...

File: README.md (hunk 41 above)
Comment (Tim2313) [reply README.md:29]: fix

File: README.md (hunk 40 above)
Comment (Tim2313) [reply README.md:26]: fix

File: README.md (hunk 37 above)
Comment (Tim2313) [reply README.md]: what does it mean **title** ?

Comment (MLGMAG) [reply README.md]: Here is file:
`myfile.txt`

The file has title/name `myfile.txt`.

title - (з укр.) назва

Comment (Tim2313) [reply README.md]: fix

File: README.md [hunk 42]
```diff
@@ -23,7 +23,18 @@ To build and compile the project, follow these steps:
 - Run the following command to build the JAR file: `mvn clean package`
 
 ## Usage
-- To start the program with the following command: `java -jar <your-project>.jar`
+Directory should have file: '<your_input>.txt'
```
Comment (MLGMAG) [inline README.md:26]: Ok...

Compile and package your program.
Place the JAR file into the separate dir with input file `test.txt`.
Populate `test.txt` with correct input.
Execute your program.
What do you get?

File: README.md [hunk 43]
```diff
@@ -23,7 +23,18 @@ To build and compile the project, follow these steps:
...
+The input file should contain only commands from section "Examples of syntax".
+The input file should be located in the same directory with the JAR file.
+Execute the program with help of the following command: `java -jar <jar_title>.jar`.
+
+You will get the following results based on the input:
+
```
Comment (MLGMAG) [inline README.md:32]: Remove the blank line.

File: README.md [hunk 44]
```diff
@@ -23,7 +23,18 @@ To build and compile the project, follow these steps:
...
+- If input file is present:
+Program will give you an error.
+
+- If there is no input file
+Program will give you an output file.
```
Comment (MLGMAG) [inline README.md:37]: Add the information about `title`, `location` of the output file.

File: README.md [hunk 45]
```diff
@@ -23,7 +23,18 @@ To build and compile the project, follow these steps:
...
-- To start the program with the following command: `java -jar <your-project>.jar`
+Directory should have file: '<your_input>.txt'
+The input file should contain only commands from section "Examples of syntax".
+The input file should be located in the same directory with the JAR file.
+Execute the program with help of the following command: `java -jar <jar_title>.jar`.
+
+You will get the following results based on the input:
+
+- If input file is present:
+Program will give you an error.
```
Comment (MLGMAG) [inline README.md:34]: Provide the example of an error.

**Review by MLGMAG — CHANGES_REQUESTED** [review]

File: README.md (hunk 42 above)
Comment (Tim2313) [reply README.md:26]: output.txt

File: README.md (hunk 45 above)
Comment (Tim2313) [reply README.md:34]: fix

File: README.md (hunk 43 above)
Comment (Tim2313) [reply README.md:32]: fix

File: README.md (hunk 42 above)
Comment (Tim2313) [reply README.md:26]: fix

File: README.md (hunk 44 above)
Comment (Tim2313) [reply README.md:37]: fix

File: README.md [hunk 46]
```diff
@@ -31,7 +31,10 @@ Execute the program with help of the following command: `java -jar <jar_title>.j
 You will get the following results based on the input:
 
 - If input file is present:
-Program will give you an error.
+Program will give you an error:
+Exception in thread "main" java.lang.ExceptionInInitializerError
+Caused by: java.lang.RuntimeException: java.io.FileNotFoundException: input.txt (No such file or directory).
+And create empty file in the directory where you execute the program: 'output.txt'.
```
Comment (MLGMAG) [inline README.md:37]: Change it to:
```
And the empty file `output.txt` will be created in current directory.
``

File: README.md [hunk 47]
```diff
@@ -31,7 +31,10 @@ Execute the program with help of the following command: `java -jar <jar_title>.j
...
 
 - If there is no input file
 Program will give you an output file.
```
Comment (MLGMAG) [inline README.md:40]: Add `location`, `title` for 'output file'.

File: README.md (hunk 42 above)
Comment (MLGMAG) [reply README.md:26]: You are wrong. 
You will get an excetion during the execution

Comment (MLGMAG) [reply README.md:26]: Why?

**Review by MLGMAG — CHANGES_REQUESTED** [review]

File: README.md (hunk 42 above)
Comment (Tim2313) [reply README.md:26]: because i need file.txt with name 'input' only

File: README.md (hunk 47 above)
Comment (Tim2313) [reply README.md:40]: fix

**Review by MLGMAG — CHANGES_REQUESTED** [review]

File: README.md [hunk 48]
```diff
@@ -23,21 +23,20 @@ To build and compile the project, follow these steps:
...
+
+- If there is no input file:
 Program will give you an error:
 Exception in thread "main" java.lang.ExceptionInInitializerError
 Caused by: java.lang.RuntimeException: java.io.FileNotFoundException: input.txt (No such file or directory).
-And create empty file in the directory where you execute the program: 'output.txt'.
-
-- If there is no input file
-Program will give you an output file.
+And the empty file `output.txt` will be created in current directory.
```
[the remark this answers is not in the document]
Comment (Tim2313) [reply README.md:39]: fix

File: README.md [hunk 49]
```diff
@@ -23,21 +23,20 @@ To build and compile the project, follow these steps:
 - Run the following command to build the JAR file: `mvn clean package`
 
 ## Usage
-Directory should have file: '<your_input>.txt'
+Directory should have file: 'input.txt'
```
[the remark this answers is not in the document]
Comment (Tim2313) [reply README.md:26]: fix

File: README.md [hunk 50]
```diff
@@ -23,21 +23,20 @@ To build and compile the project, follow these steps:
...
 The input file should be located in the same directory with the JAR file.
 Execute the program with help of the following command: `java -jar <jar_title>.jar`.
 
 You will get the following results based on the input:
-
 - If input file is present:
+Program will create and filled file in the directory where you execute the program: 'output.txt'.
+
+- If there is no input file:
 Program will give you an error:
```
[the remark this answers is not in the document]
Comment (Tim2313) [reply README.md:36]: fix

File: buildJar.bash [hunk 51]
```diff
@@ -16,5 +16,9 @@ cd $FILE
 
 jar cfm TestTaskBook.jar ../$MANIFEST_FILE ./* 
 
-mv TestTaskBook.jar ../
+cd .. 
+
+mkdir Test
+
+mv TestTaskBook.jar Test
```
Comment (MLGMAG) [inline buildJar.bash:23]: Revert changes from the `buildjar.bash` file.

**Review by MLGMAG — APPROVED** [review]

File: buildJar.bash (hunk 51 above)
Comment (Tim2313) [reply buildJar.bash]: fix

**Comment by MLGMAG** [comment]
Squash all commits into 1

**Review by MLGMAG — APPROVED** [review]

**Review by MLGMAG — APPROVED** [review]

**Pull request merged**